src/autocomplete.py

```diff
@@ -0,0 +1,27 @@
+
+class AutoCompleter(object):
+    """Create an auto compeleter."""
+
+    def __init__(self, vocab, max_completions=5):
+        """Initilizae auto completer."""
+        self.vocab = vocab
+        self.max_completions = max_completions
+        if not isinstance(self.vocab, list):
+            raise TypeError("Must enter a list of words.")
+        
+        for item in self.vocab:
+            if not isinstance(item, str):

```

Comment:
Unfortunately, this is will fail in some Python 2/3 bytes/unicode scenarios.

e.g. if in Python 2 you want to build a vocabulary which includes non-ascii characters, you need to bring the data in as `unicode`, whereas in Python 2, `str` means bytestrings.